lib/Entities/NewsletterEntity.php

```diff
@@ -257,6 +258,25 @@ public function getStatus() {
    */
   public function setStatus($status) {
     $this->status = $status;
+
+    $queues = $this->getUnfinishedQueues();
+    $newTaskStatus = null;
+    $typesWithActivation = [self::TYPE_NOTIFICATION, self::TYPE_WELCOME, self::TYPE_AUTOMATIC];
+
+    if (($status === self::STATUS_DRAFT) && in_array($this->type, $typesWithActivation)) {
+      $newTaskStatus = ScheduledTaskEntity::STATUS_PAUSED;
+    }
+    if (($status === self::STATUS_ACTIVE) && in_array($this->type, $typesWithActivation)) {
+      $newTaskStatus = ScheduledTaskEntity::STATUS_SCHEDULED;
+    }
+
+    if (!$newTaskStatus) return;
```

Comment:
I think it would be nicer to fetch the unfinished queues after this check so that we don't fetch them when they are not needed.